virtcontainers/sandbox.go

```diff
@@ -926,6 +929,102 @@ func (s *Sandbox) removeNetwork() error {
 	return s.network.remove(s, s.networkNS, s.networkNS.NetNsCreated)
 }
 
+// UpdateRoute updates the sandbox route table (e.g. for portmapping support).
+func (s *Sandbox) UpdateRoute(routes []*grpc.Route) error {
+	if err := s.agent.updateRoutes(routes); err != nil {
+		return err
+	}
+	s.networkNS.Routes = routes
+	return s.storage.storeSandboxNetwork(s.id, s.networkNS)
+}
+
+// AddNetwork adds new nic to the sandbox.
+func (s *Sandbox) AddNetwork(inf grpc.Interface) error {
+	hw, err := net.ParseMAC(inf.HwAddr)
+	if err != nil {
+		return err
+	}
+
+	var addrs []netlink.Addr
+	for _, addr := range inf.IPAddresses {
+		netlinkAddrStr := fmt.Sprintf("%s/%s", addr.Address, addr.Mask)
+		netlinkAddr, err := netlink.ParseAddr(netlinkAddrStr)
+		if err != nil {
+			return fmt.Errorf("could not parse %q: %v", netlinkAddrStr, err)
+		}
+
+		addrs = append(addrs, *netlinkAddr)
+	}
+
+	netInfo := NetworkInfo{
+		Iface: NetlinkIface{
+			LinkAttrs: netlink.LinkAttrs{
+				Name:         inf.Name,
+				HardwareAddr: hw,
+				MTU:          int(inf.Mtu),
+			},
+			Type: "",
+		},
+		Addrs: addrs,
+	}
```

Comment:
Maybe you could introduce a function translating a `grpc.Interface` structure into a `NetworkInfo` and vice-versa. This would be nice in case it needs to be reused by some other part of the code.

Reply:
Done. But the function is a little wired. Should we delete the log part?